x-pack/plugins/index_management/public/application/components/mappings_editor/lib/utils.ts

```diff
@@ -39,13 +39,23 @@ import { TreeItem } from '../components/tree';
 
 export const getUniqueId = () => uuidv4();
 
+const fieldsWithoutMultiFields: DataType[] = [
+  'flattened',
+  'join',
+  'constant_keyword',
+  'shape',
+  'geo_shape',
+  'point',
+  // @ts-expect-error aggregate_metric_double is not yet supported by the editor
+  'aggregate_metric_double',
+];
 export const getChildFieldsName = (dataType: DataType): ChildFieldName | undefined => {
-  if (dataType === 'text' || dataType === 'keyword' || dataType === 'ip') {
-    return 'fields';
+  if (fieldsWithoutMultiFields.includes(dataType)) {
+    return undefined;
   } else if (dataType === 'object' || dataType === 'nested') {
     return 'properties';
   }
-  return undefined;
+  return 'fields';
```

Comment:
Possibly a non-issue if `percolator` is added to the `fieldsWithoutMultiFields` const. However, I noticed that if I added a multi-field to `percolator`, then deleted it, it maintained any empty `fields` object and ES still threw an error. I had to delete the mapping and then recreate it to work around the issue.

<img width="922" alt="Screenshot 2024-03-07 at 1 07 34 PM" src="https://github.com/elastic/kibana/assets/5226211/1f4dddf9-c450-496b-a65e-73bedca3c2e7">

<img width="261" alt="Screenshot 2024-03-07 at 12 54 36 PM" src="https://github.com/elastic/kibana/assets/5226211/a444d7d2-a06b-497f-a4c5-6175aba2f833">

<img width="495" alt="Screenshot 2024-03-07 at 1 08 21 PM" src="https://github.com/elastic/kibana/assets/5226211/6fd8eb9e-5040-48b0-9c04-e1075b8d5834">

Reply:
Do you think it makes sense to delete the `fields` object if a multi-field is added and then deleted? I assume it's just a no-op on the ES side, but I'm not 100%.

Reply:
I asked the ES team but it doesn't seem to be a concern, but we can still open an issue for that. I'd prefer not to add it to this PR because I'm not sure that is an easy change